File: extract_streams.js
const http = require('http');

const SERVER_URL = "http://your-iptv-domain.com";
const USERNAME = "username";
const PASSWORD = "password";
const SEARCH_KEYWORD = "irishman";

const url = `${SERVER_URL}/player_api.php?username=${USERNAME}&password=${PASSWORD}&action=get_vod_streams`;

console.log("Connecting to Xtream server...");

http.get(url, (res) => {
    let data = '';
    
    res.on('data', (chunk) => {
        data += chunk;
    });
    
    res.on('end', () => {
        try {
            const movies = JSON.parse(data);
            if (!Array.isArray(movies)) {
                console.error("API response is not an array:", data.slice(0, 200));
                return;
            }
            console.log(`Successfully fetched ${movies.length} movies. Searching for '${SEARCH_KEYWORD}'...`);
            const matches = movies.filter(m => m.name && m.name.toLowerCase().includes(SEARCH_KEYWORD));
            
            if (matches.length === 0) {
                console.log(`No movie matching '${SEARCH_KEYWORD}' found.`);
                return;
            }

            console.log("\nFound matches:");
            matches.forEach(m => {
                const name = m.name;
                const stream_id = m.stream_id;
                const ext = m.container_extension || 'mp4';
                const streamUrl = `${SERVER_URL}/movie/${USERNAME}/${PASSWORD}/${stream_id}.${ext}`;
                
                console.log(`- Movie Name: ${name}`);
                console.log(`  Stream ID : ${stream_id}`);
                console.log(`  Extension : ${ext}`);
                console.log(`  Direct URL: ${streamUrl}\n`);
            });
        } catch (e) {
            console.error("Failed to parse JSON response:", e.message);
        }
    });
}).on('error', (err) => {
    console.error("HTTP request failed:", err.message);
});

File: extract_streams.py
import urllib.request
import json
import ssl

# Credentials
SERVER_URL = "http://your-iptv-domain.com"
USERNAME = "username"
PASSWORD = "password"
SEARCH_KEYWORD = "irishman"

def fetch_and_find():
    # Construct API endpoint for VOD streams
    api_url = f"{SERVER_URL}/player_api.php?username={USERNAME}&password={PASSWORD}&action=get_vod_streams"
    
    print(f"Connecting to Xtream server: {SERVER_URL}...")
    
    # Bypass SSL errors if any (for secure connections)
    ctx = ssl.create_default_context()
    ctx.check_hostname = False
    ctx.verify_mode = ssl.CERT_NONE
    
    try:
        req = urllib.request.Request(
            api_url, 
            headers={'User-Agent': 'Mozilla/5.0'}
        )
        with urllib.request.urlopen(req, context=ctx, timeout=15) as response:
            data = response.read().decode('utf-8')
            
        streams = json.loads(data)
        if not isinstance(streams, list):
            print("Error: Invalid response format from Xtream API.")
            return

        print(f"Retrieved {len(streams)} movies. Searching for '{SEARCH_KEYWORD}'...")
        
        matches = [m for m in streams if m.get('name') and SEARCH_KEYWORD in m.get('name').lower()]
        
        if not matches:
            print(f"No movie matching '{SEARCH_KEYWORD}' was found.")
            return
            
        print("\nFound matches:")
        for movie in matches:
            name = movie.get('name')
            stream_id = movie.get('stream_id')
            ext = movie.get('container_extension', 'mp4')
            
            # Construct the stream link using the same format as the app code
            stream_url = f"{SERVER_URL}/movie/{USERNAME}/{PASSWORD}/{stream_id}.{ext}"
            
            print(f"- Movie Name: {name}")
            print(f"  Stream ID : {stream_id}")
            print(f"  Extension : {ext}")
            print(f"  Direct URL: {stream_url}\n")
            
    except Exception as e:
        print(f"Error occurred: {e}")

if __name__ == "__main__":
    fetch_and_find()
